### PR TITLE
enhance: Add a config item for partition name as regexp feature and disable it by default

### PR DESCRIPTION
**Pull request by congqixia**
See also #29177
Add a config item for partition name as regexp feature and disable it by default

### CODE REVIEW

File: internal/proxy/task_search.go [hunk 1]
```diff
@@ -80,27 +79,18 @@ func getPartitionIDs(ctx context.Context, dbName string, collectionName string,
 		return nil, err
 	}
 
-	partitionsRecord := make(map[UniqueID]bool)
+	partitionsRecord := typeutil.NewSet[int64]()
 	partitionIDs = make([]UniqueID, 0, len(partitionNames))
 	for _, partitionName := range partitionNames {
-		pattern := fmt.Sprintf("^%s$", partitionName)
```
Comment (xiaofan-luan) [inline internal/proxy/task_search.go:86]: please add a config to enable regex if necessary

Comment (congqixia) [reply internal/proxy/task_search.go:86]: OK, working on it

**Review by yah01 — COMMENTED** [review]
/lgtm

**Comment by NiftyliuS** [comment]
Can you please add an explanation of how to re-enable this feature? I cant update to 2.3.4 because this breaks a filter i have on a partition key field...

**Comment by xiaofan-luan** [comment]
> In the release notes you have: To reduce resource consumption, regular expression searches in partitions have been discontinued. However, this feature can be re-enabled through configuration (see [#29154](https://github.com/milvus-io/milvus/pull/29154) for details).
> 
> Can you please add an explanation of how to re-enable this feature? I cant update to 2.3.4 because this breaks a filter i have on a partition key field...
> 
> PS: this disables automatic partitions expr

you need to set proxy.partitionNameRegexp to true

**Comment by xiaofan-luan** [comment]
In fact we don't recommend to enable this features since this might a serious performnace problem

**Comment by xiaofan-luan** [comment]
> "you need to set proxy.partitionNameRegexp to true" - you mean in the mivlus.yaml? is there an env param to override this?
> 
> Yes i get it that its not recommended, I did read the notes on performance, the problem is that currently i have milvus setup using automatic partition feature. Disabling thing means i cant filter by document_id at all, for any value.
> 
> And since i initially setup the partitions automatically using `is_partition_key` it means i have to redo the code and migrate the collection to a new one entirely...
> 
> For me this is a breaking change that came in a patch version...
> 
> Speaking of - what is the correct way to address `document_id` in my case? i intend to have 60.000 + books vectorized and i do want to be able to search all of them at once ( standard search ) as well as a search in a specific book only. I cant create partition per book, the limit is 10k partitions right?
> 
> Should i just set 128 partitions and hash the document_id to one of them and use that for search? help is much appreciated

you have to config this in milvus.yaml

**Comment by xiaofan-luan** [comment]
can if you use partition keys, then each time you should just search with one partition key right?
you can directly specify which partition key you are gonna to use and there is no need to use a regular expression

**Comment by NiftyliuS** [comment]
no it says "cant specify partion keys manually when partion key mode is enabled" i tried that, cant insert or select this way
the partitions created are the default 64 all named `_default0-63` and i cant really figure out where the document id was allocated to.

**Comment by NiftyliuS** [comment]
About workaround for the future
What if i create a hashing function  that hashes the document_id to 1-4096 and inserts it this way.
Its not ideal because i will have to filter by document_id anyway but document_id will no longer be a partition key.

hopefully this will reduce the search time, wdyt?

**Comment by NiftyliuS** [comment]
In the auto partition key mode the new flag disables searching by document_id entirely.  
It creates 64 partitions named `_default0-63` 

when not specifying partitions document_id is ignored and i get 0 results.
when specifying partitions i get an error `manually setting partitions is not allowed when using partition mode`
(i will update with the exact message once i get to the laptop)

I think you are right, setting up a new collection with manual partitions might be the way to go...

**Pull request merged**